src/CommitmentsV2/SFA.DAS.CommitmentsV2/Services/EditApprenticeshipValidationService.cs

```diff
@@ -502,6 +517,79 @@ private IEnumerable<DomainError> BuildEndDateValidationFailures(EditApprenticesh
+        {
+            switch (apprenticeshipRequest.DeliveryModel)
+            {
+                case DeliveryModel.Regular:
+                {
+                    if (apprenticeshipRequest.EmploymentEndDate != null)
+                    {
+                        yield return new DomainError(nameof(apprenticeshipRequest.EmploymentEndDate), "The employment end date cannot be set");
+                    }
+                    if (apprenticeshipRequest.EmploymentPrice != null)
+                    {
+                        yield return new DomainError(nameof(apprenticeshipRequest.EmploymentPrice), "The employment price cannot be set");
+                    }
+
+                    break;
+                }
+                case DeliveryModel.PortableFlexiJob:
+                {
+                    foreach (var failure in BuildFlexibleEmploymentPriceValidationFailures(apprenticeshipRequest))
+                    {
+                        yield return failure;
+                    }
+
+                    foreach (var failure in BuildFlexibleEmploymentDateValidationFailures(apprenticeshipRequest))
+                    {
+                        yield return failure;
+                    }
+
+                    break;
+                }
+            }
+        }
+
+        private IEnumerable<DomainError> BuildFlexibleEmploymentDateValidationFailures(EditApprenticeshipValidationRequest request)
+        {
+            if (request.EmploymentEndDate == null)
+            {
+                yield return new DomainError(nameof(request.EmploymentEndDate), "You must add the employment end date");
+            }
+            else if (request.EmploymentEndDate.Value < request.StartDate?.AddMonths(3))
+            {
+                yield return new DomainError(nameof(request.EmploymentEndDate), "This date must be at least 3 months later than the planned apprenticeship training start date");
+            }
+            if (request.EmploymentEndDate > request.EndDate)
+            {
+                yield return new DomainError(nameof(request.EmploymentEndDate), "This date must not be later than the projected apprenticeship training end date");
+            }
+        }
+
+        private IEnumerable<DomainError> BuildFlexibleEmploymentPriceValidationFailures(EditApprenticeshipValidationRequest request)
+        {
+            if (request.EmploymentPrice == null || request.EmploymentPrice <= 0)
+            {
+                yield return new DomainError(nameof(request.EmploymentPrice), "You must add the agreed price for this employment");
+            }
+
+            if (request.EmploymentPrice > Constants.MaximumApprenticeshipCost)
+            {
+                yield return new DomainError(nameof(request.EmploymentPrice), "The agreed price for this employment must be £100,000 or less");
+            }
+
+            if (request.Cost.GetValueOrDefault() <= 0)
```

Comment:
Can this condition can be removed? Cost can never be 0 when changing an apprenticeship.